src/silo/config/runtime_config.cpp

```diff
@@ -4,50 +4,183 @@
-      );
-      api_options.port = *value;
-   }
-   if (auto value = config.getInt32(ESTIMATED_STARTUP_TIME_IN_MINUTES_OPTION)) {
-      SPDLOG_DEBUG(
-         "Using {} as passed via {}: {}",
-         ESTIMATED_STARTUP_TIME_IN_MINUTES_OPTION.toString(),
-         config.configType(),
-         *value
-      );
-      const std::chrono::minutes minutes = std::chrono::minutes(*value);
+ConfigSpecification RuntimeConfig::getConfigSpecification() {
+   return {
+      .program_name = "silo api",
+      .attribute_specifications =
+         {
+            ConfigAttributeSpecification::createWithoutDefault(
+               runtimeConfigOptionKey(),
+               ConfigValueType::PATH,
+               "The path to the config file in YAML format."
+            ),
+            ConfigAttributeSpecification::createWithoutDefault(
+               defaultRuntimeConfigOptionKey(),
+               ConfigValueType::PATH,
+               "The path to config file in YAML format with default values. \n"
+               "This path will often be set by an environment variable, thus \n"
+               "providing defaults to a silo in a specific environment (e.g. Docker)."
+            ),
+            ConfigAttributeSpecification::createWithDefault(
+               dataDirectoryOptionKey(),
+               ConfigValue::fromPath(DEFAULT_OUTPUT_DIRECTORY),
+               "The path to the directory with the data files (output from preprocessing)."
+            ),
+            ConfigAttributeSpecification::createWithDefault(
+               apiPortOptionKey(),
+               ConfigValue::fromUint16(8081),
+               "The port number on which to listen for incoming HTTP connections."
+            ),
+            ConfigAttributeSpecification::createWithDefault(
+               apiMaxConnectionsOptionKey(),
+               ConfigValue::fromInt32(64),
+               "The maximum number of concurrent connections accepted at any time."
+            ),
+            ConfigAttributeSpecification::createWithDefault(
+               apiParallelThreadsOptionKey(),
+               ConfigValue::fromInt32(4),
+               "The number of worker threads."
+            ),
+            ConfigAttributeSpecification::createWithoutDefault(
+               apiEstimatedStartupTimeOptionKey(),
+               ConfigValueType::UINT32,
+               "Estimated time in minutes that the initial loading of the database takes. \n"
+               "As long as no database is loaded yet, SILO will throw a 503 error. \n"
+               "This option allows SILO to compute a Retry-After header for the 503 response."
+            ),
+            ConfigAttributeSpecification::createWithDefault(
+               queryMaterializationOptionKey(),
+               ConfigValue::fromUint32(10000),
+               "Above how many records in a result set the result rows are to be constructed\n"
+               "lazily (by streaming)."
+            ),
```

Comment:
"materialization" is used at least in Python to collect iterators into actual, "materialized", lists. That's why I used the term.

With another couple tweaks I find your help text suggestion good:
"if a query results in fewer rows, the query result will be collected in memory before sending it to the client. If it affects more rows, it will be streamed by constructing the result items lazily."